### PR TITLE
Upgrade open telemetry package dependencies for Azure Monitor OpenTelemetry Exporter

### PR DESCRIPTION
**Pull request by vishweshbankwar**
1) Upgrading OpenTelemetry to use 1.1.0-beta4
2) Updgrading OpenTelemetry.Extensions.Hosting, OpenTelemetry.Instrumentation.AspNetCore and OpenTelemetry.Instrumentation.Http to 1.0.0-rc5

### CODE REVIEW

**Review by rajkumar-rangaraj — APPROVED** [review]

File: eng/Packages.Data.props [hunk 1]
```diff
@@ -98,7 +98,7 @@
     <PackageReference Update="System.IdentityModel.Tokens.Jwt" Version="5.4.0" />
 
     <!-- OpenTelemetry dependency approved for Azure.Monitor.OpenTelemetry.Exporter package only -->
-    <PackageReference Update="OpenTelemetry" Version="1.0.1"  Condition="'$(MSBuildProjectName)' == 'Azure.Monitor.OpenTelemetry.Exporter'" />
+    <PackageReference Update="OpenTelemetry" Version="1.1.0-beta4"  Condition="'$(MSBuildProjectName)' == 'Azure.Monitor.OpenTelemetry.Exporter'" />
```
Comment (reyang) [inline eng/Packages.Data.props:101]: @pakrym what's the recommended practice for integration with the upcoming release of OpenTelemetry?
1. Create a separate branch? (any branching naming rule?)
2. Use main branch for anything that is not yet GA'ed?

**Review by reyang — APPROVED** [review]
LGMT given the exporter is still preview.

Let's see if @pakrym would suggest putting this in a feature branch.

File: eng/Packages.Data.props (hunk 1 above)
Comment (pakrym) [reply eng/Packages.Data.props:101]: Using the main branch is OK while the exporter is in preview.

**Review by pakrym — APPROVED** [review]

**Pull request merged**